bhyve.pyx

```diff
@@ -1,4 +1,139 @@
-cimport bhyve
+cimport vmmapi
+cimport cpuset
+cimport types
+cimport vmm
 
+import enum
+import os
 
+from libc.errno cimport errno
+from libc.stdlib cimport free
 
+
+class Error(enum.IntEnum):
+    OPEN_FAILED = 100
+    STATS_FETCH_FAILED = 101
+    RETRIEVE_ACTIVE_VCPU_FAILED = 102
+
+
+class BhyveException(RuntimeError):
+    def __init__(self, code, message):
+        super(BhyveException, self).__init__(f'[Errorno {code}] {message}')
+        self.code = code
+
+cdef class VM:
+
+    cdef str vm_name
+    cdef vmmapi.vmctx* vm
+
+    def __cinit__(self, str vm_name_p, bint create_vm=0):
+        self.vm_name = vm_name_p
+        vm_name_bytes = vm_name_p.encode()
+        cdef char* vm_name = vm_name_bytes
+
+        if create_vm:
+            with nogil:
+                code = vmmapi.vm_create(vm_name)
```

Comment:
from libvmmapi vm_create only creates the guest name space inside vmm device. I understand this module is under development, just to let you aware of it.